src/core/QCheck2.ml

```diff
@@ -750,6 +750,11 @@ module Gen = struct
   let add_shrink_invariant (p : 'a -> bool) (gen : 'a t) : 'a t =
     fun st -> gen st |> Tree.add_shrink_invariant p
 
+  let without_shrinker (gen: 'a t) : 'a t =
+    make_primitive
+      ~gen:(fun st -> gen st |> Tree.root)
```

Comment:
```suggestion
      ~gen:generate1
```

I think it's equivalent

Reply:
I don't think so, as I suspect `generate1` will ignore cook up its own `Random.State`:
```ocaml
  let generate1 ?(rand=RS.make_self_init()) (gen : 'a t) : 'a =
    gen rand |> Tree.root
```
I think this means the root for a particular run will change if you annotate with `without_shrinker`.
On the other hand, that might be a good test to add... :wink:

Reply:
Oh yes, my suggestion was bad indeed. I meant:
```ocaml
make_primitive
      ~gen:(fun rand -> generate1 ~rand gen)
      ...
```